### PR TITLE
azure-artifacts-credprovider: init at 1.0.9

### PR DESCRIPTION
**Pull request by anpin**
## Description of changes

The Azure Artifacts Credential Provider enables dotnet, NuGet.exe, and MSBuild to interactively acquire credentials for Azure Artifacts feeds. 

https://github.com/microsoft/artifacts-credprovider

## Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/az/azure-artifacts-credprovider/package.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+, icu
+, openssl
+, zlib
+, curl
+, dotnet-sdk_6
+, fetchFromGitHub
+}:
+buildDotnetModule rec {
+  inherit dotnet-sdk_6;
+  version = "1.0.9";
```
Comment (FliegendeWurst) [inline pkgs/by-name/az/azure-artifacts-credprovider/package.nix:16]:
```suggestion
  version = "1.3.0";
```
New version released by now

File: pkgs/by-name/az/azure-artifacts-credprovider/package.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  nugetDeps = ./deps.nix;
+  passthru.updateScript = ./update.sh;
+  patchPhase = ''
+    sed -i 's|<TargetFrameworks>.*</TargetFrameworks>|<TargetFramework>net8.0</TargetFramework>|' Build.props
+  '';
+  meta = with lib; {
+    homepage = "https://github.com/microsoft/artifacts-credprovider";
+    description = "Azure Artifacts Credential Provider";
+    license = licenses.mit;
+    maintainers = with maintainers; [ anpin ];
```
Comment (FliegendeWurst) [inline pkgs/by-name/az/azure-artifacts-credprovider/package.nix:29]:
```suggestion
  meta = {
    homepage = "https://github.com/microsoft/artifacts-credprovider";
    description = "Azure Artifacts Credential Provider";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ anpin ];
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/az/azure-artifacts-credprovider/package.nix (hunk 2 above)
Comment (anpin) [reply pkgs/by-name/az/azure-artifacts-credprovider/package.nix]: force pushed proposed changes

**Review by FliegendeWurst — APPROVED** [review]

**Comment by anpin** [comment]
Thanks!

**Pull request merged**